File: app/src/main/java/com/mgaetan89/showsrage/model/ShowStat.java
package com.mgaetan89.showsrage.model;

import com.google.gson.annotations.SerializedName;

import java.io.Serializable;
import java.util.Map;

public class ShowStat implements Serializable {
	private static final long serialVersionUID = 7360065212498675713L;

	private int archived;
	private Map<String, Integer> downloaded;
	private int failed;
	private int ignored;
	private int skipped;
	private Map<String, Integer> snatched;
	@SerializedName("snatched_best")
	private int snatchedBest;
	private int subtitled;
	private int total;
	private int unaired;
	private int wanted;

	public int getArchived() {
		return this.archived;
	}

	public Map<String, Integer> getDownloaded() {
		return this.downloaded;
	}

	public int getFailed() {
		return this.failed;
	}

	public int getIgnored() {
		return this.ignored;
	}

	public int getSkipped() {
		return this.skipped;
	}

	public Map<String, Integer> getSnatched() {
		return this.snatched;
	}

	public int getSnatchedBest() {
		return this.snatchedBest;
	}

	public int getSubtitled() {
		return this.subtitled;
	}

	public int getTotal() {
		return this.total;
	}

	public int getTotalDone() {
		if (this.downloaded == null) {
			return this.archived;
		}

		Integer total = this.downloaded.get("total");

		if (total == null) {
			return this.archived;
		}

		return this.archived + total;
	}

	public int getTotalPending() {
		if (this.snatched == null) {
			return this.snatchedBest;
		}

		Integer total = this.snatched.get("total");

		if (total == null) {
			return this.snatchedBest;
		}

		return this.snatchedBest + total;
	}

	public int getUnaired() {
		return this.unaired;
	}

	public int getWanted() {
		return this.wanted;
	}
}

File: app/src/main/kotlin/com/mgaetan89/showsrage/model/ShowStat.kt
package com.mgaetan89.showsrage.model

import com.google.gson.annotations.SerializedName

data class ShowStat(
		val archived: Int = 0,
		val downloaded: Map<String, Int>? = null,
		val failed: Int = 0,
		val ignored: Int = 0,
		val skipped: Int = 0,
		val snatched: Map<String, Int>? = null,
		@SerializedName("snatched_best") val snatchedBest: Int = 0,
		val subtitled: Int = 0,
		val total: Int = 0,
		val unaired: Int = 0,
		val wanted: Int = 0
) {
	fun getTotalDone(): Int {
		if (this.downloaded == null) {
			return this.archived
		}

		val total = this.downloaded["total"] ?: return this.archived

		return this.archived + total
	}

	fun getTotalPending(): Int {
		if (this.snatched == null) {
			return this.snatchedBest
		}

		val total = this.snatched["total"] ?: return this.snatchedBest

		return this.snatchedBest + total
	}
}
